posydon/popsyn/normalized_pop_mass.py

```diff
@@ -107,30 +108,74 @@ def fraction_simulated(f_bin, m_1, m_2, m_min, m_max,
         alpha2 = 2.35
         alpha3 = 2.35
     else:
-        Pwarn("Scheme not included yet: primary_mass_scheme="
-              f"{kwargs['primary_mass_scheme']}, secondary_mass_scheme"
-              f"={kwargs['secondary_mass_scheme']}", "UnsupportedModelWarning")
-        return np.nan, np.nan, np.nan
-#        raise ValueError("Scheme not included yet")
+        raise ValueError("Scheme not included yet")
 
-    f_bin = 0.7
+    f_bin_nature = 0.7 #This parameter represents the preferred fraction of binary systems within your population, allowing users to input any value ranging from 0.1 to 1.
+    f_bin_simulated = kwargs['binary_fraction_const']
+    
     m_min = 0.01
     m_max = 200.0
     m_a = kwargs['primary_mass_min']
     m_b = kwargs['primary_mass_max']
 
     f0 = 1/(quad(imf_part_1, m_min, m_1, args=(m_min, alpha1))[0]
-            + quad(imf_part_2, m_1, m_2, args=(m_1, m_min, alpha1, alpha2))[0]
-            + quad(imf_part_3, m_2, m_max, args=(m_1, m_2, m_min, alpha1,
-                                                 alpha2, alpha3))[0])
-
-    f_corr = (fraction_simulated(f_bin, m_1, m_2, m_min,
-                                 m_max, alpha1, alpha2, alpha3)
-              * mean_mass_simulated(alpha3, m_a, m_b)
-              / mean_mass_of_binary(f0, f_bin, m_1, m_2, m_min, m_max,
-                                    alpha1, alpha2, alpha3))
-
-    f_model = fraction_simulated(f_bin, m_1, m_2, m_min, m_max,
-                                 alpha1, alpha2, alpha3)
-
-    return initial_ZAMS_TOTAL_mass / f_corr, f_corr, f_model
+                + quad(imf_part_2, m_1, m_2, args=(m_1, m_min, alpha1, alpha2))[0]
+                + quad(imf_part_3, m_2, m_max, args=(m_1, m_2, m_min, alpha1, alpha2, alpha3))[0])
+
+    
+    if (f_bin_simulated == 1) & (f_bin_nature != 0): #when you have modeled only binary stars
+       
+
+        f_corr_binaries = (fraction_simulated(f_bin_nature, m_1, m_2, m_min, m_max, alpha1, alpha2, alpha3)
+                  * mean_mass_simulated(alpha3, m_a, m_b, f_bin_simulated)
+                  / mean_mass_of_binary(f0, f_bin_nature, m_1, m_2, m_min, m_max,
+                                        alpha1, alpha2, alpha3))
+
+        underlying_total_mass=initial_ZAMS_TOTAL_binaries/f_corr_binaries
+
+        f_corr_single_stars=0
+
+    if (f_bin_simulated != 1): #when you have modeled both binary and single stars) 
+        if (f_bin_nature == 0): #you want the underlying mass for a population consisting of only single stars
+
+            
+
+            f_corr_single_stars = (fraction_simulated(1, m_1, m_2, m_min, m_max, alpha1, alpha2, alpha3)
```

Comment:
Why is the first argument here `1` and not `f_bin_nature`?

Reply:
In this case, our simulations model both binary systems and single stars, but the desired (in nature) population consists solely of single stars (f_bin_nature=0). So we have modeled the following fraction of all systems (1-fbin_nature)=1.